src/libbson/src/bson/memory.c

```diff
@@ -231,6 +232,79 @@ bson_aligned_alloc0(size_t alignment /* IN */, size_t num_bytes /* IN */)
 }
 
 
+/*
+ *--------------------------------------------------------------------------
+ *
+ * bson_array_alloc --
+ *
+ *       Allocates memory for an array of objects.
+ *
+ *       Libbson does not try to handle OOM conditions as it is beyond the
+ *       scope of this library to handle so appropriately.
+ *
+ * Parameters:
+ *       @num_elems: The number of objects to allocate.
+ *       @elem_size: The size of each object in bytes.
+ *
+ * Returns:
+ *       A pointer if successful; otherwise abort() is called and this
+ *       function will never return.
+ *
+ *--------------------------------------------------------------------------
+ */
+
+void *
+bson_array_alloc(size_t num_elems /* IN */, size_t elem_size /* IN */)
+{
+   void *mem = NULL;
+   size_t num_bytes;
+   BSON_ASSERT(!mlib_mul(&num_bytes, num_elems, elem_size));
```

Comment:
Because of an implementation quirk, `num_bytes` must be initialized before passing its address to `mlib_mul`. (It doesn't matter what value.) (This is the cause of the current task failures on MSVC.)

```suggestion
   size_t num_bytes = 0;
   BSON_ASSERT(!mlib_mul(&num_bytes, num_elems, elem_size));
```